src/iceberg_functions/iceberg_multi_file_reader.cpp

```diff
@@ -71,15 +71,14 @@ idx_t IcebergMultiFileList::GetTotalFileCount() {
 }
 
 unique_ptr<NodeStatistics> IcebergMultiFileList::GetCardinality(ClientContext &context) {
-	auto total_file_count = IcebergMultiFileList::GetTotalFileCount();
-
-	if (total_file_count == 0) {
-		return make_uniq<NodeStatistics>(0, 0);
+	idx_t cardinality = 0;
```

Comment:
I think this needs to call GetFile just like `GetTotalFileCount` was doing, because we need to make sure we've looked through all manifests at this point